### PR TITLE
WASM: Support complex type in arrays

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards https://github.com/lcompilers/lpython/issues/1485.

### CODE REVIEW

File: src/libasr/codegen/asr_to_wasm.cpp [hunk 1]
```diff
@@ -75,16 +75,20 @@ enum RT_FUNCS {
...
     cur_mem_loc = 0,
     tmp_reg_i32 = 1,
     tmp_reg_i64 = 2,
     tmp_reg_f32 = 3,
-    tmp_reg_f64 = 4,
-    GLOBAL_VARS_CNT = 5
+    tmp_reg2_f32 = 4,
+    tmp_reg_f64 = 5,
+    tmp_reg2_f64 = 6,
+    GLOBAL_VARS_CNT = 7
```
Comment (certik) [inline src/libasr/codegen/asr_to_wasm.cpp:91]: I think you don't have to number them, since the default numbering is 0, 1, 2, ...

**Review by certik — APPROVED** [review]
Looks good.

File: src/libasr/codegen/asr_to_wasm.cpp (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/libasr/codegen/asr_to_wasm.cpp:91]: Thank you for reviewing my pull request. I understand that the default numbering for items is 0, 1, 2, and so on. However, I chose to be explicit with the numbering for clarity purposes. Would you like me to remove the numbering, or is it alright to leave it as is?

Comment (certik) [reply src/libasr/codegen/asr_to_wasm.cpp:91]: I don't think it matters, whatever you find more readable.

**Pull request merged**